lightning/mydump/parser.go

```diff
@@ -175,44 +259,82 @@ func (parser *ChunkParser) ReadRow() error {
 	type state byte
 
 	const (
-		// the state after reading "VALUES"
-		stateRow state = iota
-		// the state after reading the table name, before "VALUES"
+		// the state after "INSERT INTO" before the column names or "VALUES"
+		stateTableName state = iota
+
+		// the state while reading the column names
 		stateColumns
+
+		// the state after reading "VALUES"
+		stateValues
+
+		// the state while reading row values
+		stateRow
 	)
 
 	row := &parser.lastRow
-	st := stateRow
+	st := stateValues
```

Comment:
why not initial to `stateTableName`?

Reply:
Because an INSERT statement may have multiple rows, and this method only read one.

```sql
INSERT INTO `table` (`a`, `b`) VALUES (1, 2)   -- first ReadRow() call parse until here
,(3, 4)                                        -- second ReadRow() call parse until here
;
```